### ISSUE TITLE
[Performance]: eager leaf construction pays the ~5.6us backend session open (from_tensor_in ~12us for 8 elements)

### ISSUE THREAD

**Issue by lingrui96** [tensor4all/tenferro-rs#1704]
## What happened?

`EagerTensor::from_tensor_in` costs ~12 us for an 8-element leaf, of which
~5.6 us is opening a backend session that does no work.

`new_leaf` (`crates/tenferro-ad/src/eager.rs`) makes the incoming tensor
contiguous through the execution session:

```rust
let semantic_tensor = ctx
    .with_execution_session(|session| {
        session.to_contiguous_read(TensorRead::from_tensor(&tensor))
    })?
```

`with_execution_session` goes through `lock_backend()` and
`with_backend_session`, so every leaf construction pays the per-call session
open that #1667 identified for op dispatch — arbiter acquire, condvar signalling,
session construction, managed-session install.

Measured at `b5a106be` (current `main`), release, 1 thread, macOS arm64,
3000 iterations after warm-up:

| | us |
|---|---|
| `Tensor::from_vec_col_major` (8 elements) | 0.54 |
| `with_execution_session`, **empty closure** | **5.59** |
| `with_execution_session` + `to_contiguous_read` | 7.69 |
| `EagerTensor::from_tensor_in` (whole leaf) | 12.23 |

So 5.59 us is the floor for entering a session at all, before any work; the
contiguous read of 8 elements adds 2.1 us, and the remaining ~4.5 us is the
`RetainedValue` / `TracedTensor` / `AdValueRecord` construction.

The cost does not follow the data. Constructing an `IdxTensor` through
`tensor4all-core` (which wraps `EagerTensor`) measures 13.5 us at 8 elements
with one index, 13.5 us at 8 elements with three indices, and 17.3 us at 256
elements with eight indices. It is also flat in the number of live tensors
(15.1 / 14.5 / 12.7 / 13.1 us with 0 / 100 / 1000 / 10000 tensors alive), which
rules out registry growth.

## What did you expect?

Constructing a leaf from an already-contiguous host tensor should not require a
backend session. `from_tensor_in` passes `requires_grad: false` and no trace, so
nothing about the leaf needs the backend at construction time; making the read
contiguous is a check on an already-contiguous buffer.

## Relationship to #1667

#1667 reported the per-call session open at ~11-13 us for eager **op dispatch**
and is closed. The floor is now 5.59 us, so that work halved it, but it did not
remove it — and the framing there was op dispatch, so leaf construction paying
the same floor does not appear to have been in scope. This is that second path,
not a regression of the first.

## Minimal reproducer

`crates/tenferro-ad/benches/`, `cargo bench -p tenferro-ad`:

```rust
use tenferro_ad::{AdContext, EagerRuntime, EagerTensor};
use tenferro_cpu::CpuBackend;
use tenferro_tensor::{Tensor, TensorRead};

let ad = AdContext::builder().build().unwrap();
let ctx = EagerRuntime::with_cpu_backend_and_ad_context(CpuBackend::default(), &ad).unwrap();
let dims = vec![2usize, 2, 2];
let data = vec![1.0f64; 8];

// ~5.6 us, doing nothing.
let _ = ctx.with_execution_session(|_s| ()).unwrap();

// ~12.2 us.
let native = Tensor::from_vec_col_major(dims.clone(), data.clone()).unwrap();
let _ = EagerTensor::from_tensor_in(native, ctx.clone()).unwrap();
```

## Why this matters downstream

`tensor4all-core`'s `IdxTensor` wraps every tensor in an `EagerTensor`, so this
is paid on every tensor any tensor4all code constructs. In
`tensor4all-treetn`'s `TreeTNCachedEvaluator`, one batched evaluation builds
roughly four tensors per directed message across 17-31 messages, so a single
`evaluate_batched` call spends 800-1500 us in session opens and AD record
construction, for tensors whose arithmetic is a handful of flops.

Measured effect on the caller: on a chain, `tensor4all-treeaci` reaches the same
accuracy at the same or lower rank as the `TensorTrain` implementation for no
more function evaluations, but costs ~44-57 us per evaluation against ~0.5-0.6
us. The fitted slope of evaluation time against bond dimension is 0.25, against
the 2 or more the contraction arithmetic would predict — the cost is per-call
overhead, not work. The `TensorTrain` path avoids it because
tensor4all-rs#621 inlined its inner product rather than going through tensor
operations at all.

## Affected area

Eager dispatch (leaf construction), CPU backend, `tenferro-ad`.
